sqlite-migrations/src/main/kotlin/com/squareup/sqlite/migrations/ObjectDifferDatabaseComparator.kt

```diff
@@ -5,6 +5,8 @@ import de.danielbechler.diff.node.DiffNode
 
 object ObjectDifferDatabaseComparator : DatabaseComparator<CatalogDatabase> {
 
+  var circularReferenceExceptionLogger: ((String) -> Unit)? = null
```

Comment:
This basically becomes static state. Can we switch this object to a class and force callers to create and manage an instance?

Reply:
@JakeWharton Yes, it also makes sense for me, I just didn't want to change existing API, but because it used only here, I will convert it to a class, thanks for confirming

Reply:
Fixed and rebased